src/ast/ast_validations.c

```diff
@@ -472,6 +447,18 @@ static AST_Validation _ValidateInlinedProperties(const cypher_astnode_t *props)
 		return AST_INVALID;
 	}
 
+	uint prop_count = cypher_ast_map_nentries(props);
+	for(uint i = 0; i < prop_count; i++) {
+		const cypher_astnode_t *prop_val = cypher_ast_map_get_value(props, i);
+		cypher_astnode_type_t prop_type = cypher_astnode_type(prop_val);
+		if(prop_type == CYPHER_AST_PATTERN_PATH) {
+			// Encountered query of the form:
+			// MATCH (a {prop: ()-[]->()}) RETURN a
```

Comment:
funny parser considers this as a valid query, I guess `()-[]->()` is an expression which evaluates to a PATH value, do we have a test covering this case?